test/json_unittest.cc

```diff
@@ -112,7 +232,24 @@ TEST(TrivialToStringTest, StringOneChar) {
 TEST(TrivialParseTest, ArrayEmpty) {
   GuardJsonException([](){
     json::value v = json::Parser::FromString("[]");
-    EXPECT_TRUE(v->As<json::Array>()->empty());
+    EXPECT_EQ(json::ArrayType, v->type());
+
+    EXPECT_FALSE(v->Is<json::Null>());
+    EXPECT_FALSE(v->Is<json::Boolean>());
+    EXPECT_FALSE(v->Is<json::Number>());
+    EXPECT_FALSE(v->Is<json::String>());
+    EXPECT_TRUE(v->Is<json::Array>());
+    EXPECT_FALSE(v->Is<json::Object>());
+
+    EXPECT_THROW(v->As<json::Null>(), json::Exception);
+    EXPECT_THROW(v->As<json::Boolean>(), json::Exception);
+    EXPECT_THROW(v->As<json::Number>(), json::Exception);
+    EXPECT_THROW(v->As<json::String>(), json::Exception);
+    const auto& arr = v->As<json::Array>();
```

Comment:
move this line down - to just before the expect_true

Reply:
They're in that order deliberately, as that's the order of complexity of the json types